### DISCUSSION TITLE
Do I need some kind of custom bot protection service when I use Vercel?

### DISCUSSION THREAD

**Discussion by offchan42** [vercel/community#5651] [Feature Question] [answered]
### Question

I noticed that Vercel already protects against DDoS to some extent according to this article: https://vercel.com/docs/security/ddos-mitigation

Do I still need to use some services like ReCAPTCHA or Cloudflare to protect against automatic tool that spam the website?
If I do, which service do you suggest me using?
Specifically I'm building an AI art generator app and provide free tier so I'm afraid that some users will create a bunch of accounts and spam the site to death.

**Comment by leerob** [comment, marked as answer]
Hey, wanted to follow up here.

We just launched **[Attack Challenge Mode](https://vercel.com/changelog/prevent-malicious-traffic-with-attack-challenge-mode-for-vercel-firewall)** –an extra layer of defense for DDoS and malicious traffic, free for all plans. When enabled, the Vercel Firewall will serve a challenge page to help mitigate attacks.

Hope this helps!
